src/main/java/com/surrealdb/refactor/driver/parsing/ResultParser.java

```diff
@@ -0,0 +1,63 @@
+package com.surrealdb.refactor.driver.parsing;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.surrealdb.refactor.exception.UnhandledProtocolResponse;
+import com.surrealdb.refactor.types.QueryResult;
+
+public class ResultParser {
+    // this method parses the JSON element regardless of whether it is a single element
+    // or an array of elements
+
+    public ResultParser() {}
+
+    public QueryResult[] parseResultMessage(JsonElement resultMessage) {
+        QueryResult[] processedOuterResults;
+
+        // checks to see if result is a JSON element, if not throw exception
+        if (resultMessage.isJsonObject()) {
+            JsonArray outerResultArray;
+
+            if (!resultMessage.isJsonArray()) {
+
+                JsonObject jsonObject = new JsonObject();
+                jsonObject.addProperty("status", "ok");
+                jsonObject.addProperty("time", System.currentTimeMillis());
+                jsonObject.add("result", resultMessage.getAsJsonObject());
+
+                // add element to an array if it is not an array
+                outerResultArray = new JsonArray();
+                outerResultArray.add(jsonObject);
+
+            } else if (resultMessage.isJsonArray()) {
+
+                outerResultArray = resultMessage.getAsJsonArray();
+            } else {
+                // this puts in the else if for the if it is a jsonArray; however,
+                // I am unsure of when this portion of the code would run
+                throw new RuntimeException(
+                        "https://github.com/surrealdb/surrealdb.java/issues/75\n "
+                                + "The Outer Result Array came up for false for (1) is not an"
+                                + " Array and (2) is an Array");
```

Comment:
We don't need URLs, and the message can be clearer :) When users (or driver developers) see the exception, they need to be able to see what happened conveniently - implementation details tend not to be important, only explaining the assumptions and why things are having to "break" with an exception.

```suggestion
                throw new RuntimeException("The input json message was neither an object or an array");
```